docs/deployment.md

```
@@ -0,0 +1,156 @@
++    app: magicapi-gateway
++  ports:
++    - protocol: TCP
++      port: 80
++      targetPort: 3000
++  type: ClusterIP
++```
+
+## Production Considerations
+
+### Security Checklist
+- [ ] Enable TLS/SSL
```

Markdownlint

Expected: plus; Actual: dash
Unordered list style

(MD004, ul-style)